packages/gatsby-source-drupal/src/gatsby-node.js

```diff
@@ -102,7 +104,96 @@ exports.sourceNodes = async (
     return
   }
 
-  const drupalFetchActivity = reporter.activityTimer(`Fetch data from Drupal`)
+  fastBuilds = fastBuilds || false
+  if (fastBuilds) {
+    let lastFetched = 0
+    if (
+      store.getState().status.plugins &&
+      store.getState().status.plugins[`gatsby-source-drupal`] &&
+      store.getState().status.plugins[`gatsby-source-drupal`].lastFetched
+    ) {
+      lastFetched = store.getState().status.plugins[`gatsby-source-drupal`]
+        .lastFetched
+    }
+
+    const drupalFetchIncrementalActivity = reporter.activityTimer(
+      `Fetch incremental changes from Drupal`
+    )
+    let requireFullRebuild = false
+
+    drupalFetchIncrementalActivity.start()
+
+    try {
+      // Hit fastbuilds endpoint with the lastFetched date.
+      const data = await axios.get(
+        `${baseUrl}/gatsby-fastbuilds/sync/${lastFetched}`,
+        {
+          auth: basicAuth,
+          headers,
+          params,
+        }
+      )
+
+      if (data.data.status === -1) {
+        reporter.info(`Unable to pull incremental data changes from Drupal`)
+        setPluginStatus({ lastFetched: data.data.timestamp })
```

Comment:
Should teh `lastFetched` indeed be updated if the fetch failed? Or does the `-1` status code mean there simply isn't anything to update?

Reply:
Great question. I think an additional comment makes a lot of sense here. The -1 status code means the timestamp has either expired (too long since last sync) or they have never synced in the past. We track the lastfetched time (for the next run) and then trigger a full build.